packages/core/src/integration.ts

```diff
@@ -155,3 +164,54 @@ function findIndex<T>(arr: T[], callback: (item: T) => boolean): number {
 
   return -1;
 }
+
+/**
+ * Generate a full integration function from a simple function.
+ * This will ensure to add the given name both to the function definition (as id),
+ * as well as to the integration return value.
+ */
+export function makeIntegrationFn<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Fn extends (...rest: any[]) => Partial<IntegrationFnResult>,
+>(name: string, fn: Fn): ((...rest: Parameters<Fn>) => ReturnType<Fn> & { name: string }) & { id: string } {
+  const patchedFn = addNameToIntegrationFnResult(name, fn);
+
+  return Object.assign(patchedFn, { id: name });
```

Comment:
There is no need for an object.assign in here. It is slow. We can just call patchedFn.id = name

Reply:
This is basically done to make TS happy, but I can also "force" it via a type conversion! This is the cleanest, from a TS perspective, that I could find 😬